query-engine/schema/src/input_types.rs

```diff
@@ -69,6 +71,14 @@ impl<'a> InputObjectType<'a> {
         self.tag.as_ref()
     }
 
+    pub fn container(&self) -> Option<&ParentContainer> {
+        self.container.as_ref()
+    }
+
+    pub fn set_container(&mut self, container: ParentContainer) {
+        self.container = Some(container);
+    }
```

Comment:
nit: I would make this method polymorphic so you can pass models and composite types directly without calling `.into()` at call site:
```suggestion
    pub fn set_container(&mut self, container: impl Into<ParentContainer>) {
        self.container = Some(container.into());
    }
```

Reply:
Oh that's a cool "trick" - wasn't aware of that - thanks! 🎉

Reply:
Note that impl trait in the argument position is more or less syntax sugar for an anonymous generic, i.e., this is equivalent to:
```rust
pub fn set_container<C: Into<ParentContainer>>(&mut self, container: C) {
    self.container = Some(container.into());
}
```
This means it's monomorphized and uses static dispatch at compile time, so it's efficient but can blow up the code size real quick. It's perfect for small functions like this which are likely to be inlined anyway but you might not want to use it in large functions. A pattern you might see in some libraries is to split the generic and non-generic parts into separate functions, and provide a generic facade that does the conversions and calls the large non-generic function.